controllers/certificatesigningrequests_controller.go

```diff
@@ -102,54 +100,35 @@ func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, err
 	}
 
-	return ctrl.Result{}, r.reconcileCSR(certificateSigningRequest)
+	return ctrl.Result{}, r.validateAndApprove(certificateSigningRequest)
 }
 
-// reconcileCSR kicks off CSR validation and approval. In case of update conflicts, the approval process is wrapped
-// with retry logic using the Update --> Get --> Retry Update pattern. Retry will occur until specified timeout
-func (r *certificateSigningRequestsReconciler) reconcileCSR(req *certificates.CertificateSigningRequest) error {
-	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
-		err := r.validateAndApprove(req)
-		if err == nil {
-			return true, nil
-		}
-		if !k8sapierrors.IsConflict(err) {
-			// If the error is not an update conflict, return error immediately
-			return false, err
+// validateAndApprove handles the CSR validation and approval. Process wrapped in retry logic case of update conflicts
+func (r *certificateSigningRequestsReconciler) validateAndApprove(req *certificates.CertificateSigningRequest) error {
+	err := k8sretry.RetryOnConflict(k8sretry.DefaultBackoff, func() error {
```

Comment:
When you are doing this retry you need to fetch the CSR each retry, else you are acting on stale information. see function documentation [here](https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict)

I would think you'd need to pass in the namespaced name to this function.
the get being done within the `Reconcile` func should be removed and that logic should also be captured within the function passed into `RetryOnConflict`

Reply:
Great catch, definitely deleted then ignored the re-fetching. I'll rework this.

Reply:
Done below

Reply:
Nice catch @sebsoto 👍🏽